### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by xakep666**
This PR adds 2 workflows for Actions:
* Go Test. It runs tests (only unit-tests currently) and uploads coverage report to Codecov if tests runs successfully. This workflow runs on pull request with target branch 'master' and push to branch 'master'.
* Release. It runs `goreleaser` with preparations (docker login) for each new tag.

Following secrets must be added:
* `DOCKER_USERNAME`, `DOCKER_PASSWORD` - DockerHub credentials
* `HOMEBREW_TAP_TOKEN` - Token for Homebrew tap.
* `CODECOV_TOKEN` - Token for Codecov

This PR should solve problems like #394

Also this PR removes unused CI configurations: `travis-ci` and `appveyor`.

### CODE REVIEW

**Comment by xakep666** [comment]
@LandonTClipp you mentioned github actions port in #394 so probably you can review this.

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (grongor) [inline .github/workflows/testing.yml:11]: I think Windows is still supported, so you should include it here too. Maybe including macOS isn't a bad idea either.

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
```
Comment (grongor) [inline .github/workflows/testing.yml:18]: You should also include the "tip" version, as it was before (the latest Go version)

**Review by grongor — CHANGES_REQUESTED** [review]

File: .github/workflows/testing.yml (hunk 2 above)
Comment (xakep666) [reply .github/workflows/testing.yml:18]: Looks good but may lead to bloat: (Linux, Windows, Mac)×(1.16,tip)
Splitting to different parallel workflows may help, will try this

Comment (grongor) [reply .github/workflows/testing.yml:18]: I think you can use matrix builds for that. Here is an example how I did it with my library and I think it works well: https://github.com/grongor/panicwatch/blob/master/.github/workflows/ci.yaml

Comment (xakep666) [reply .github/workflows/testing.yml:18]: Thanks, didn'k know about matrix feature in Actions

**Comment by LandonTClipp** [comment]
This is great, thanks for taking the time. I will look at this tonight and approve if it looks good.

**Comment by LandonTClipp** [comment]
It appears I don't have permissions to modify secrets on this repo, I need to contact the owner so I can get added.

**Comment by LandonTClipp** [comment]
I was able to access secrets through the API. With some minor changes, the build worked wonderfully. Homebrew, dockerhub, GH release page all working. Thanks a ton :1st_place_medal:

**Comment by LandonTClipp** [comment]
We'll want to implement the testing matrix for different version but in the immediate term this gets us out the door.

**Pull request merged**